bot.py

```diff
@@ -38,14 +31,12 @@ def roll_dice(dice_count, dice_size):
         await asyncio.sleep(5)
         await client.send_message(message.channel, 'Done sleeping')
     elif message.content.startswith("!roll"):  # Example: "!roll 3d6".
-	# TODO: include "XdY + Z", "XdY + ZdN", hits/dicepools, and exploding dice.
-        FUNC_INPUT = message.content[6:].split("d")
-	DICEROLL = roll_dice(int(FUNC_INPUT[0]), int(FUNC_INPUT[1]))
-	RESULTS = "Rolled a " + str(sum(DICEROLL)) + ", " +str(DICEROLL)
+	DICEROLL = preparse(message.content[6:])
+	RESULTS = "Rolled a " + str(DICEROLL[0]) + ", " +str(DICEROLL[1])
         await client.send_message(message.channel, RESULTS))
     elif message.content.startswith("!map"):  # Example: "!map 400, 800".
         DIMENSIONS = message.content[5:].split(", ")
         save_to_image(delake(blur(walk(build(int(DIMENSIONS[0]), int(DIMENSIONS[1]))))), OUTPUT_LOCATION="C:\\Users\\My Dell\\Desktop\\", OUTPUT_FILE_NAME="Map")  # ! The Pillow module, and by extension mapgen, requires a file destination by default. I'll need help setting it up/changing it to store to a database. Alternatively, the bot could only bother storing one map at a time, long enough to post it, leaving the more long-term storage to Discord and/or a secondary bot.
-	# await client.post_image(IMAGE_LOCATION)  # ! Dummy function. I'll be amazed if post_image() is an actual thing.
+	# await client.post_image(IMAGE_LOCATION)  # ! Dummy function. I'll be surprised if post_image() is an actual thing.
```

Comment:
I'm not sure if bots are allowed to upload images to Discord. I'll check when I'm not passing out.

One thing we can do instead is to store the image on our server, serve it over HTTPS, and let the bot reply in chat with a link to the resource on the server.